core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -526,38 +395,42 @@ class WorkspaceService(
         }
       }
     )
-
-  private def maybeLoadMcWorkspace(workspaceContext: Workspace): Future[Option[WorkspaceDescription]] =
-    workspaceContext.workspaceType match {
-      case WorkspaceType.McWorkspace =>
-        Future(Option(workspaceManagerDAO.getWorkspace(workspaceContext.workspaceIdAsUUID, ctx)))
-      case WorkspaceType.RawlsWorkspace => Future(None)
-    }
+  }
 
   private def gatherWorkflowsToAbortAndSetStatusToAborted(workspaceName: WorkspaceName, workspaceContext: Workspace) =
-    dataSource.inTransaction { dataAccess =>
-      for {
-        // Gather any active workflows with external ids
-        workflowsToAbort <- dataAccess.workflowQuery.findActiveWorkflowsWithExternalIds(workspaceContext)
-
-        // If a workflow is not done, automatically change its status to Aborted
-        _ <- dataAccess.workflowQuery.findWorkflowsByWorkspace(workspaceContext).result.map { workflowRecords =>
-          workflowRecords
-            .filter(workflowRecord => !WorkflowStatuses.withName(workflowRecord.status).isDone)
-            .foreach { workflowRecord =>
-              dataAccess.workflowQuery.updateStatus(workflowRecord, WorkflowStatuses.Aborted) { status =>
-                if (config.trackDetailedSubmissionMetrics)
-                  Option(
-                    workflowStatusCounter(workspaceSubmissionMetricBuilder(workspaceName, workflowRecord.submissionId))(
-                      status
+    dataSource
+      .inTransaction { dataAccess =>
+        for {
+          // Gather any active workflows with external ids
+          workflowsToAbort <- dataAccess.workflowQuery.findActiveWorkflowsWithExternalIds(workspaceContext)
+
+          // If a workflow is not done, automatically change its status to Aborted
+          _ <- dataAccess.workflowQuery.findWorkflowsByWorkspace(workspaceContext).result.map { workflowRecords =>
+            workflowRecords
+              .filter(workflowRecord => !WorkflowStatuses.withName(workflowRecord.status).isDone)
+              .foreach { workflowRecord =>
+                dataAccess.workflowQuery.updateStatus(workflowRecord, WorkflowStatuses.Aborted) { status =>
+                  if (config.trackDetailedSubmissionMetrics)
+                    Option(
+                      workflowStatusCounter(
+                        workspaceSubmissionMetricBuilder(workspaceName, workflowRecord.submissionId)
+                      )(
+                        status
+                      )
                     )
-                  )
-                else None
+                  else None
+                }
               }
-            }
-        }
-      } yield workflowsToAbort
-    }
+          }
+        } yield workflowsToAbort
+      }
+      .recover { case t: Throwable =>
+        logger.warn(
```

Comment:
This recover error block was moved here to reduce clutter in the `deleteWorkspace` function.